rportfwd segfaults if nothing listen at port at remote system

I tried the new rportfwd feature, and it's awesome. I did notice a bug however.

If you establish a reverse portforward like so:
rportfwd add -b 0.0.0.0:8000 -r 127.0.0.1:8000

and test a reverse connection with something like a netcat listener:
nc -lvp 8000

Then use something like a browser to test the portforward:
└─$ nc -lvp 8000
listening on [any] 8000 ...
connect to [127.0.0.1] from localhost [127.0.0.1] 36634
GET / HTTP/1.1

Then hit ctrl + c to kill the netcat listener, promptly forget this and refresh the browser at the victim machine you get this:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xd7c444]

goroutine 119 [running]:
github.com/bishopfox/sliver/server/handlers.createReverseTunnelHandler(0xc000146a10, {0xc000b28300, 0x2e0, 0x300})
        github.com/bishopfox/sliver/server/handlers/sessions.go:296 +0x604
github.com/bishopfox/sliver/server/handlers.tunnelDataHandler(0xc000146a10, {0xc000b28300, 0x2e0, 0x300})
        github.com/bishopfox/sliver/server/handlers/sessions.go:129 +0x405
github.com/bishopfox/sliver/server/c2.handleSliverConnection.func2.2()
        github.com/bishopfox/sliver/server/c2/mtls.go:116 +0x37
created by github.com/bishopfox/sliver/server/c2.handleSliverConnection.func2
        github.com/bishopfox/sliver/server/c2/mtls.go:115 +0x538

--- a/implant/sliver/rportfwd/portfwd.go
+++ b/implant/sliver/rportfwd/portfwd.go
@@ -165,7 +165,9 @@ func (p *ChannelProxy) HandleConn(src net.Conn) {
 		log.Printf("[portfwd] Closing tunnel %d (%s)", tunnel.ID, reason)
 		// {{end}}
 		tunnel := p.Conn.Tunnel(tunnel.ID)
-		p.Conn.RemoveTunnel(tunnel.ID)
+		if tunnel != nil {
+			p.Conn.RemoveTunnel(tunnel.ID)
+		}
 		src.Close()
 		cancelContext()
 	}

--- a/server/handlers/sessions.go
+++ b/server/handlers/sessions.go
@@ -244,6 +244,7 @@ func createReverseTunnelHandler(implantConn *core.ImplantConnection, data []byte
 			Data: tunnelClose,
 		}
 		cancelContext()
+		return nil
 	}
 
 	if conn, ok := dst.(*net.TCPConn); ok {

